File: python/organisms/plants/guarana.py
from gui.logger import Logger
from organisms.plants.plant import Plant
from organisms.species import Species


class Guarana(Plant):
    def __init__(self, pos):
        super().__init__(pos)
        self.species = Species.GUARANA

    def collision(self, other, attacking, grid):
        other.strength += 3
        Logger.log(other.species.name + ' gained 3 strength and ')
        return False

File: java/pl/edu/pg/mygame/organisms/plants/Guarana.java
package pl.edu.pg.mygame.organisms.plants;

import pl.edu.pg.mygame.Grid;
import pl.edu.pg.mygame.World;
import pl.edu.pg.mygame.organisms.Organism;
import pl.edu.pg.mygame.organisms.Species;

import java.awt.*;

public class Guarana extends Plant {
    public Guarana(Point p) {
        super(p);
        this.species = Species.GUARANA;
    }

    @Override
    public boolean collision(Organism other, boolean attacking, Grid grid) {
        other.setStrength(other.getStrength() + 3);
        World.message_panel.showMessage(other.getSpecies() + " gained 3 strength and ");
        return false;
    }
}
